app/navigation/navigator.py

```diff
@@ -20,8 +26,23 @@ def _valid_destination(self, destination):
         :returns: True if valid, False otherwise
```

Comment:
This function is a membership check; is the string representing a step in our flow a member of our known steps. We can thus rewrite it as an element in a set check:

Something like this (zip just creates a new list from two lists, perhaps should append the custom_steps list). 

```python
def _valid_destination(self, destination):
    RESERVED_STEPS = ['introduction', 'summary', 'thank-you']
    custom_steps = self._schema.get_steps() # Assumed this function exists somehow.
    step_group = zip(RESERVED_STEPS, custom_steps)
    return destination in step_group # This is a membership check that returns True or False dependent on result.
```